### PR TITLE
Codecov setup

### PR DESCRIPTION
**Pull request by mhmdtshref**
Setup codecov

### CODE REVIEW

**Comment by mhmdtshref** [comment]
> I can't seem to run code coverage - can you help me out?
> I am on the branch and have npm i in both the root and client directories.
> when i run `npm run coverage` I get an error:
> 
> Screen Shot 2019-05-13 at 11 37 36

There's a new directory `.nyc_output` you should add to the root directory. It's the folder for code coverage reports. So, by default it's in the gitignore file

**Comment by mhmdtshref** [comment]
> Can you explain what you did in each file and why?
@orjwan98 

For the package.json file, i've added the required packages..
For travis, I've added the package that travis should install, also the script that travis should run.

**Comment by teenie-quaggard** [comment]
@founders-programme-2/howl has anyone been able to run codecov? I have now created a directory in my root called .nyc_output but I still get an error when I run ```npm run coverage```
Screen Shot 2019-05-13 at 15 42 52

File: package.json [hunk 1]
```diff
@@ -3,13 +3,14 @@
...
     "dev": " npm run client && nodemon ./src/server.js ",
-    "start": "node ./src/server.js", 
+    "start": "node ./src/server.js",
     "eslint": "eslint ./src",
-    "eslint:fix": "eslint --fix ./src", 
-    "heroku-postbuild": "cd client && npm i && npm run build"
+    "eslint:fix": "eslint --fix ./src",
+    "heroku-postbuild": "cd client && npm i && npm run build",
+    "coverage": "nyc report --reporter=json && codecov -f coverage/coverage-final.json"
   },
```
[the remark this answers is not in the document]
Comment (mhmdtshref) [reply package.json]: You don't know how I'm happy for this help, thank you very much @Oliversw and @teenie-quaggard for this help!

File: package.json [hunk 2]
```diff
@@ -3,13 +3,14 @@
...
     "client": " cd client  && npm run build",
     "dev": " npm run client && nodemon ./src/server.js ",
-    "start": "node ./src/server.js", 
+    "start": "node ./src/server.js",
     "eslint": "eslint ./src",
-    "eslint:fix": "eslint --fix ./src", 
-    "heroku-postbuild": "cd client && npm i && npm run build"
+    "eslint:fix": "eslint --fix ./src",
+    "heroku-postbuild": "cd client && npm i && npm run build",
+    "coverage": "nyc report --reporter=json && codecov -f coverage/coverage-final.json"
```
[the remark this answers is not in the document]
Comment (mhmdtshref) [reply package.json]: I've changed this with one initial test! thanks!

File: package.json [hunk 3]
```diff
@@ -29,6 +30,7 @@
     "eslint": "^5.16.0",
     "eslint-config-airbnb-base": "^13.1.0",
     "eslint-plugin-import": "^2.17.2",
-    "nodemon": "^1.19.0"
+    "nodemon": "^1.19.0",
+    "nyc": "^14.1.1"
```
[the remark this answers is not in the document]
Comment (mhmdtshref) [reply package.json]: nyc has been uninstalled!

File: client/package.json [hunk 4]
```diff
@@ -24,7 +24,7 @@
   },
   "scripts": {
     "start": "react-scripts start",
-    "build": "react-scripts build",
+    "build": "SKIP_PREFLIGHT_CHECK=true react-scripts build",
```
Comment (Oliversw) [inline client/package.json]: I'd love to talk about what the ```SKIP_PREFLIGHT_CHECK=true``` does at some point! Maybe at tomorrow's standup

**Pull request merged**